pkg/api/rest/delete.go

```diff
@@ -69,14 +69,13 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 		// only a shorter grace period may be provided by a user
 		if options.GracePeriodSeconds != nil {
 			period := int64(*options.GracePeriodSeconds)
-			if period > *objectMeta.DeletionGracePeriodSeconds {
-				return false, true, nil
+			newDeletionTimestamp := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
```

Comment:
That's not the guarantee that we designed the graceful deletion path for, so I'm not in favor of making a change like that right now without a longer discussion.  The shorter grace period duration must be recorded because an external client may not observe the first request before starting the second.  The timestamp is advisory - the grace period is authoritative.